include/pybind11/pytypes.h

```diff
@@ -1085,6 +1085,17 @@ class str : public object {
     // NOLINTNEXTLINE(google-explicit-constructor)
     str(const std::string &s) : str(s.data(), s.size()) { }
 
+#ifdef PYBIND11_HAS_STRING_VIEW
+    // NOLINTNEXTLINE(google-explicit-constructor)
+    str(std::string_view s) : str(s.data(), s.size()) { }
+
+# ifdef PYBIND11_HAS_U8STRING
+    // NOLINTNEXTLINE(google-explicit-constructor)
+    str(std::u8string_view s) : str(s.data(), s.zie()) { }
```

Comment:
Is this a typo?
```suggestion
    str(std::u8string_view s) : str(s.data(), s.size()) { }
```
Can't find any reference to `.zie` in the std::library. If it is a typo and our test suite isn't picking it up that is a bit concerning and indicates that the constructor isn't ever compiled / is shadowed by another constructor and is never therefore called or compiled, right?

Reply:
@henryiii Another reason why we should look into adding codecoverage to our CI even if only for gcc / clang.